services/src/main/java/org/keycloak/broker/oidc/mappers/ClaimToUserSessionNoteMapper.java

```diff
@@ -0,0 +1,123 @@
+package org.keycloak.broker.oidc.mappers;
+
+import org.keycloak.broker.oidc.KeycloakOIDCIdentityProviderFactory;
+import org.keycloak.broker.oidc.OIDCIdentityProviderFactory;
+import org.keycloak.broker.provider.BrokeredIdentityContext;
+import org.keycloak.models.Constants;
+import org.keycloak.models.IdentityProviderMapperModel;
+import org.keycloak.models.IdentityProviderSyncMode;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.UserModel;
+import org.keycloak.provider.ProviderConfigProperty;
+import org.keycloak.util.JsonSerialization;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class ClaimToUserSessionNoteMapper extends AbstractClaimMapper {
+
+    private static final String CLAIM_PROPERTY_NAME = "claims";
+    private static final String[] COMPATIBLE_PROVIDERS = {KeycloakOIDCIdentityProviderFactory.PROVIDER_ID,
+            OIDCIdentityProviderFactory.PROVIDER_ID};
+
+    private static final List<ProviderConfigProperty> CONFIG_PROPERTIES = new ArrayList<>();
+
+    private static final Set<IdentityProviderSyncMode> IDENTITY_PROVIDER_SYNC_MODES =
+            new HashSet<>(Arrays.asList(IdentityProviderSyncMode.values()));
+
+    static {
+        ProviderConfigProperty claimsProperty = new ProviderConfigProperty();
+        claimsProperty.setName(CLAIM_PROPERTY_NAME);
+        claimsProperty.setLabel("Claims");
+        claimsProperty.setHelpText(
+                "Name of the claims to search for in the token. You can reference nested claims using a '.', i.e. 'address.locality'. To use dot (.) literally, escape it with backslash (\\.)");
```

Comment:
Yes, should be plural "Names".

Multiple claims have to be separated by `##` in the storage, to be displayed as multiple list entries in the UI. In the UI, the double ## won't be displayed. So the _##_ is an API and storage detail - imho it does not make sense to add this to a UI help text.

Reply:
Ah, ok. I have not checked the UI on this one. Thanks @danielFesenmeyer !

Reply:
One additional note: Is it possible to specify which `token` is meant here in this tooltip? For example something like:
`...for in the ID Token (eventually UserInfo if applicable or eventually access token if the Identity Provider is Keycloak)`. (I hope I am correct here, not 100% sure actually...:-) ). Or is it too long text? WDYT guys?

Reply:
@mposolda I just had a look together together @artur-baltabayev. The real logic in https://github.com/keycloak/keycloak/blob/7b58783255fb4059c125b168c7ec0bd68cb17778/services/src/main/java/org/keycloak/broker/oidc/mappers/AbstractClaimMapper.java#L75 is to take claim from access token if IDP is Keycloak OIDC, then from ID token if not exists and then from userinfo endpoint if still not exists. That's pretty hard to get into a tooltip. Also, the current formulation is pretty much copy&paste from all other IDP mappers, they also just talk about the `token`. It might be good to improve the wording, but I would rather do it consistently.

Reply:
@artur-baltabayev @sschu Agree, Thanks for pointing this! Resolving this tooltip